### PR TITLE
Support https; Nicer errors; Moar emojis; Reject on any error other 'FetchError: network timeout'

### PR DESCRIPTION
**Pull request by adambraimbridge**
Looking good to go.

### CODE REVIEW

**Comment by adambraimbridge** [comment]
Seems to be okay ... 
image

**Comment by adambraimbridge** [comment]
If there are no errors: It floats!

image

**Comment by adambraimbridge** [comment]
Update:
 - [ ] Moar emojis! 😻
 - [ ] If there's a "real" error, don't wait two minutes to time out.
 - [ ] De-duplication of final error message.

File: tasks/float.js [hunk 1]
```diff
@@ -60,7 +60,6 @@ function task (opts) {
 		log.art.canoe();
 		log.success('IT FLOATS!');
 	}).catch(function (err) {
-		log.error('Man overboard', err);
```
Comment (simonplend) [inline tasks/float.js]: :+1:

File: lib/wait-for-ok.js [hunk 2]
```diff
@@ -5,23 +5,31 @@ module.exports = function (url) {
 		let timeout;
 		let checker;
 		function checkGtg () {
-			console.log(`polling: ${url}`); // eslint-disable-line no-console
+			console.log(`⏳ polling: ${url}`); // eslint-disable-line no-console
 			fetch(url, { timeout: 2000, follow: 0 })
 				.then(function (response) {
 					if (response.ok) {
-						console.log(`poll ${url} ok`); // eslint-disable-line no-console
+						console.log(`⌛️ ${url} ok!`); // eslint-disable-line no-console
```
Comment (simonplend) [inline lib/wait-for-ok.js]: How about a ✔️ emoji when it's ok...

File: lib/wait-for-ok.js [hunk 3]
```diff
@@ -5,23 +5,31 @@ module.exports = function (url) {
...
 						clearTimeout(timeout);
 						clearInterval(checker);
 						resolve();
 					} else {
-						console.log(`poll ${url} not ok`); // eslint-disable-line no-console
+						console.log(`⌛️ ${url} not ok`); // eslint-disable-line no-console
```
Comment (simonplend) [inline lib/wait-for-ok.js]: ...and a ❌ for when they're not.

I figure these are easy to scan for in output to determine success or failure and similar to the output that test runners like mocha and n-test provide.

:thinking:

File: lib/wait-for-ok.js [hunk 4]
```diff
@@ -5,23 +5,31 @@ module.exports = function (url) {
...
+					}
+				})
+				.catch(error => {
+					if (error.type && error.type === 'request-timeout') {
+						console.log(`👋 Hey, ${url} doesn't seem to be responding yet, so there's that. You're amazing, by the way. I don't say that often enough. But you really are.`); // eslint-disable-line no-console
+					}
+					else {
+						reject(`😿 ${url} Error: ${error}`);
+						clearInterval(checker);
```
Comment (simonplend) [inline lib/wait-for-ok.js]: Good :cat:ch!

**Pull request merged**